core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1004,9 +1004,10 @@ private static void tryWriteTempFile(Path path) throws IOException {
             Path resolve = path.resolve(".es_temp_file");
             try {
                 Files.createFile(resolve);
-                Files.deleteIfExists(resolve);
             } catch (IOException ex) {
                 throw new IOException("failed to write in data directory [" + path + "] write permission is required", ex);
+            } finally {
+                Files.deleteIfExists(resolve);
```

Comment:
can we make sure not to delete a file that wasn't created and hide the original exception?

we typically do this:
```
bool success = false;
try {
  do something
  success = true;
} finally {
  if (success) {
    clean up
  }
}
```